gateway/webhook/webhookTransformer.go

```diff
@@ -45,11 +50,53 @@ func (bt *batchWebhookTransformerT) markResponseFail(reason string) transformerR
 	return resp
 }
 
+func (bt *batchWebhookTransformerT) transformUsingPlugin(plugin plugins.Plugin, payload []byte, sourceType string) transformerBatchResponseT {
+	transformStart := time.Now()
+	var events []map[string]interface{}
+	err := json.Unmarshal(payload, &events)
+	if err != nil {
+		bt.stats.failedStat.Count(len(events))
```

Comment:
Can we really count the `events` if the unmarshalling failed? :thinking: 
I'm afraid this might end up as a noop call like `Count(0)`.

Reply:
fixed this.

Reply:
Maybe a `failedBatch` metric can be useful so that we can `.Increment()` it. Wdyt?
Otherwise we won't have visibility about those failures in our metrics (unless we do in some other way?).

Reply:
you mean whole batch is failed? that may not happening with plugin implementation.

Reply:
What I meant is that earlier you were bumping a metric if you couldn't unmarshal the payload. Now it's not there anymore so we don't have any visibility on that, right?
Given that and given that given that "N events = batch" I proposed something like `failedBatch.Increment()`. If you don't want to call it batch please propose a different name but it's a technicality. What I'm trying to say is that we don't have a metric there anymore :slightly_smiling_face:

Reply:
Maybe we can have `pluginTransformationSuccess` and `pluginTransformationFailure` that we can increment only once per method call. And then keep also the `receivedStat` with `receivedStat.Count(len(events))`. Wdyt?

Reply:
per event level we can add we can add pluginTransformationSuccess or pluginTransformationFailure, but since we are not doing any batch operation for all events so I think we need to skip failed batch. In reality the failedBatch should never trigger as we are generating that code, unless there is not test case for it.

Reply:
Updated can you see if that make sense